src/context/context.tsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+import { Dispatch, SetStateAction, createContext, useState } from "react";
```

Comment:
Thanks for setting this to a typescript file. I was struggling with some of the types here - ProductDetails, etc.